ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -121,7 +131,19 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 				if err != nil {
 					return clusterCRs, err
 				}
-
+				if kind == "BareMetalHost" {
+					// Check for note level BIOS configuration
+					if siteConfigTemp.Spec.Clusters[clusterId].Nodes[ndId].BiosConfigMapRef.FilePath != "" {
```

Comment:
If you like, I think you should be able to adapt a multi-level lookup scheme I added; see `node.CrTemplateSearch` and maybe we can just use similar logic to lookup the BiosConfigMapRef too, by sharing some of the code and the algorithm?

Reply:
Sure, I will make changes accordingly.